pyproject.toml

```diff
@@ -26,13 +26,20 @@ requires-python = ">=3.9"
 
 dependencies = [
     "qiskit>=1.2, <3",
-    "numpy>=1.26",
+    "numpy>=2.0",
```

Comment:
Need to bump this because we use ``npuint64 << 1`` in the new functionality, which wasn't supported until numpy 2.0